interview_questions.md

```diff
@@ -11,127 +11,916 @@ and Rails, grouped by category.
 ## Web Development
 
 - What is meant by frontend and backend?
+  ```
+  The part of a website that user interacts with directly is termed as front end. It is also referred to as the 'client side' of the application.
+  Backend is 'server side' of the website. It stores and arranges data, and also makes sure everything on the client-side of the website works fine. It is the part of the website that you cannot see and interact with.
+  ```
+
 - What are web frameworks?
+  ```
+  A web framework or web application framework is a software framework that is designed to support the development of web applications including web services, web resources, and web APIs.
+  ```
+
 - What are some common frontend web frameworks?
+  ```
+  Frontend or CSS Frameworks: React, Angular, Vue, etc.
+  ```
+
 - What is meant by responsive web design?
+  ```
+  A responsive website has a fluid and flexible layout which adjusts according to screen size. Basically, your website will look great and work well on a laptop, a tablet, and a mobile phone's browser.
+  ```
+
 - What are some common backend web frameworks?
+  ```
+  Backend Frameworks : Laravel Framework, Flask Framework, Django, etc.
+  ```
+
 - Explain MVC architecture and its benefits?
+  ```
+  The Model-View-Controller (MVC) is an architectural pattern that separates an application into three main logical components: the model, the view, and the controller.
```

Comment:
You should define what is meant by the terms model, view and controller further.

Reply:
Ohk👍